build/tasks/AddArchiveReferences.cs

```diff
@@ -52,12 +52,28 @@ public override bool Execute()
             foreach (var package in archiveArtifacts)
             {
                 var packageName = package.ItemSpec;
-                var packageVersion = string.Equals(packageName, "Microsoft.AspNetCore.All", StringComparison.OrdinalIgnoreCase) ?
-                    MetaPackageVersion :
-                    buildArtifacts
-                        .Single(p => string.Equals(p.PackageInfo.Id, packageName, StringComparison.OrdinalIgnoreCase))
-                        .PackageInfo.Version.ToString();
 
+                string packageVersion;
+
+                if (string.Equals(packageName, "Microsoft.AspNetCore.All", StringComparison.OrdinalIgnoreCase)
+                   || string.Equals(packageName, "Microsoft.AspNetCore.App", StringComparison.OrdinalIgnoreCase))
+                {
+                    packageVersion = MetaPackageVersion;
+                }
+                else
+                {
+                    try
+                    {
+                        packageVersion = buildArtifacts
+                            .Single(p => string.Equals(p.PackageInfo.Id, packageName, StringComparison.OrdinalIgnoreCase))
+                            .PackageInfo.Version.ToString();
+                    }
+                    catch (InvalidOperationException)
+                    {
+                        Log.LogError($"Missing Package: {packageName} from artifacts archive.");
```

Comment:
I have a slight preference for changing the call from `.Single` to a `.Where()`. InvalidOperationException could thrown in two conditions: 0 packages or more than 1 package with that ID.

Reply:
So that we can differentiate between package not found and too many packages found?

Reply:
yeah, that's where i was going with that. Update the error to either say "Missing package" or "Multiple packages" found. I think missing package is the most likely cause of the error right now, but that could always change.